modules/core/src/main/java/org/apache/synapse/mediators/eip/splitter/IterateMediator.java

```diff
@@ -117,70 +129,143 @@ public boolean mediate(MessageContext synCtx) {
         }
 
         try {
-            // get a copy of the message for the processing, if the continueParent is set to true
-            // this original message can go in further mediations and hence we should not change
-            // the original message context
-            SOAPEnvelope envelope = MessageHelper.cloneSOAPEnvelope(synCtx.getEnvelope());
 
-            synCtx.setProperty(id != null ? EIPConstants.EIP_SHARED_DATA_HOLDER + "." + id :
-                               EIPConstants.EIP_SHARED_DATA_HOLDER, new SharedDataHolder());
+            // check whether expression contains jsonpath or xpath and process according to it
+            if (expression != null && expression instanceof SynapseJsonPath) {
```

Comment:
Redundant null check

Reply:
This was added to prevent null expressions getting inside the else block. If we remove null check, null expressions will get executed inside the else block.

Reply:
Still, a null expression would go inside the else block, right? This will evauluate to true only if the expression is not null.